File: scripts/mission_node_stub.py
#!/usr/bin/env python
import rospy
import actionlib 
from mission_planner.msg import MissionPlannerAction, MissionPlannerGoal, MissionPlannerResult, MissionPlannerFeedback

class MissionNodeStub:
    _result = MissionPlannerResult()

    def __init__(self):
        self._action_name = rospy.get_name()
        self._as = actionlib.SimpleActionServer(self._action_name, MissionPlannerAction, execute_cb=self.actionCallback, auto_start = False)
        self._as.start()


    def actionCallback(self, goal):
        rospy.loginfo("%s, actionCallback called!" % self._action_name)

        rospy.sleep(3.)		# sleep for 1 sec

        rospy.loginfo('%s: send return!' % self._action_name)
        rospy.loginfo('-------------------------------')
        self._as.set_succeeded(self._result)


if __name__ == '__main__':
    rospy.init_node('mission_stub_node', anonymous=True)	
    try: 
        mission_node_stub = MissionNodeStub()
        rospy.spin()	
		
    except rospy.ROSInterruptException:
        print(error)
        pass			
			

File: src/mission_node_stub.cpp
#include "mission_node_stub/mission_node_stub.h"
#include <string>

using namespace std;

MissionNodeStub::MissionNodeStub(char* action_name)
    : as_(nh_, string(action_name), boost::bind(&MissionNodeStub::actionCallback, this, _1), false),  action_name_(action_name)
{
    nh_ = ros::NodeHandle("~");

	as_.start();
}

void MissionNodeStub::actionCallback(const state_cpp_msg::MissionPlannerGoalConstPtr& goal)
{
    ROS_INFO("%s, actioncallback called!", action_name_);

    ros::Duration(1).sleep();       // sleep for 1 sec

    ROS_INFO("%s, send return!", action_name_);
    ROS_INFO("---------------------------");

    state_cpp_msg::MissionPlannerResult result;
	as_.setSucceeded(result);
}
